plugins/mediatek-scaler/fu-mediatek-scaler-device.c

```diff
@@ -571,8 +587,11 @@ fu_mediatek_scaler_device_commit_firmware(FuMediatekScalerDevice *self,
 					  GError **error)
 {
 	guint16 sum16 = 0;
-	if (!fu_input_stream_compute_sum16(stream, &sum16, error))
-		return FALSE;
+	g_autoptr(GBytes) fw = NULL;
+
+	fw = fu_input_stream_read_bytes(stream, 0, G_MAXSIZE, error);
+	sum16 = fu_sum16_bytes(fw);
```

Comment:
ew? why this rather than `fu_input_stream_compute_sum16`? Doing `fu_input_stream_read_bytes()` as one big allocation uses significantly more RSS than just reading from a file descriptor.

Reply:
fixed.